x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java

```diff
@@ -41,8 +41,10 @@
         }
         if ("true".equals(property)) {
             AUTOSCALING_FEATURE_FLAG_REGISTERED = true;
-        } else if ("false".equals(property) || property == null) {
+        } else if ("false".equals(property)) {
             AUTOSCALING_FEATURE_FLAG_REGISTERED = false;
+        } else if (property == null) {
+            AUTOSCALING_FEATURE_FLAG_REGISTERED = null;
```

Comment:
I suggest leaving it as a non-nullable `boolean` and saying this here.

```suggestion
            AUTOSCALING_FEATURE_FLAG_REGISTERED = Build.CURRENT.isSnapshot();
```

Then this flag exactly reflects whether the feature flag is registered or not.

Reply:
There's two issues with that suggestion:
 - my intention is that the field is aligned with the value of the system property `es.autoscaling_feature_flag_registered`, it's merely a `Boolean` representation of the value of that system property
 - the intention of using `Autoscaling#isSnapshot` in determining whether or not to register the setting is so that it can be overridden in tests; if we tie whether or not to register the setting directly to `Build#Current#isSnapshot` as you suggest, we lose that ability